graphFunctions.js

```diff
@@ -0,0 +1,296 @@
+// Function to reorganize the data to limit parameter constraints
+// Inputs: data, an Array of sub-arrays (defined below),
+//			limit, Integer value of maximum allowed devices,
+//			setting, Integer value depending on structure of input
+//			data
+function reduceLength(data, limit, setting){
+	var newData = [];
+	// Make sure the data needs to be modified
+	if (data.length > limit){
+		var sum = 0;
+		// determine input data structure
+		// setting === 0, data is in [Name, value] format
+		// setting === 1, data is in [Rank, Name, value, percentage] format
+		if (setting <= 1){
+			for (var i = 0; i < limit; i++){
+				// Consider using a splice here since the structure
+				// remains the same but for loop still needed for
+				// sum
+				var value = parseInt(data[i][1 + setting]);
+				newData.push([data[i][0 + setting],value]);
+				sum += value;
+			}
+		}
+		else {
+			console.log("reduceLength, invalid setting value used");
+		}
+		// Create Other Devices entry
+		newData.push(["Other Devices", totalNumUsers - sum]);
+	}
+	return newData;
+}
+
+// Function to parse data by selected devices

```

Comment:
Parsing but I use the plotGraph start for the function name so there's no confusion that this is used for the plotGraph function later. Should I change this to parseSelectedDevices?